### DISCUSSION TITLE
Scope vs all apps modules

### DISCUSSION THREAD

**Discussion by TurboLed** [LSPosed/LSPosed#1149] [Q&A] [answered]
Why isn't there an "All apps" scope ? I saw comments on a previous (locked) issue asking to be able to apply a module to all apps, saying those modules that require access to all apps are inferior. I disagree with that.
The whole point of XPosed is to be able to hook into any Android code. A lot of the code run inside the instance of each app without going through specific Android system apps. 
For example, how can someone make a module to change the color of the status bar for every app? The status bar color is controlled by each app via setStatusBarColor and the app itself draws its own status bar background. 
Now with LSPosed it's impossible to have a module to force a certain status bar color (it was removed from GravityBox most likely for the same reason).
I'm sure there are many other examples. Limiting a module scope to only certain apps or system packages is certainly not the way the original XPosed vision was!
Please add support for an "all apps" scope or something that allows us to hook certain API calls from within any app that gets running!

**Comment by JasonKhew96** [comment, marked as answer]
There is a reason that module scope exist
- safety
- performance

> Now with LSPosed it's impossible to have a module to force a certain status bar color (it was removed from GravityBox most likely for the same reason).

![image](https://user-images.githubusercontent.com/29531167/133970504-d83197ec-e3b9-46b8-a0f6-285d669cbe7f.png)

> Please add support for an "all apps" scope or something that allows us to hook certain API calls from within any app that gets running!

There is an API to let xposed module set or get their scope on the way (https://github.com/LSPosed/LSPosed/issues/9)

**Reply by ghost** [reply]
> There is an API to let xposed module set or get their scope on the way (#9)

This is the solution to all these requests. Many issue has FR "all apps" or similar. 
A module can receive intent of ACTION_PACKAGE_ADDED or ACTION_PACKAGE_INSTALL and choose if enable scope for that app.
